src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -58,7 +62,62 @@ static void initConfiguration(SecureCommunication& secureCommunication, const nl
         if (config.at("ssl").contains("certificate_authorities") &&
             !config.at("ssl").at("certificate_authorities").empty())
         {
-            caRootCertificate = config.at("ssl").at("certificate_authorities").front().get_ref<const std::string&>();
+            std::vector<std::string> filePaths =
+                config.at("ssl").at("certificate_authorities").get<std::vector<std::string>>();
+
+            if (filePaths.size() > 1)
+            {
+                std::string caRootCertificateContentMerged;
+
+                for (const auto& filePath : filePaths)
+                {
+                    if (!std::filesystem::exists(filePath))
+                    {
+                        throw std::runtime_error("The CA root certificate file: '" + filePath + "' does not exist.");
+                    }
+
+                    std::ifstream file(filePath);
+                    if (!file.is_open())
+                    {
+                        throw std::runtime_error("Could not open CA root certificate file: '" + filePath + "'.");
+                    }
+
+                    caRootCertificateContentMerged.append((std::istreambuf_iterator<char>(file)),
+                                                          std::istreambuf_iterator<char>());
+                }
+
+                // Once all the files are read, we write the content into final file.
+                caRootCertificate = "/var/ossec/var/certs/root-ca-merged.pem";
```

Comment:
As @MiguelazoDS stated, lets check if we can make this path relative

Reply:
If the folder exists the result is successfully

```console
2024/07/18 13:16:58 indexer-connector[8114] indexerConnector.cpp:115 at initConfiguration(): DEBUG: All CA files merged into 'var/certs/root-ca-merged.pem' successfully.
```

Reply:
Changes done!